blog/2022-07-09-git-contribute-guide.mdx

```diff
@@ -0,0 +1,187 @@
+git clone https://github.com/擁有者/專案名字
+```
+
+<Adsense/>
+
+![](/img/git-contribution-guide/CloneProject.png)
+
+然後專案就會被複製到電腦上您指定的位置了，但是請先別急，請先透過以下指令添加一個原本專案的 git 數據庫（在此我命名該數據庫為 upstream）：
+
+```shell showLineNumbers
+git remote add upstream 原本專案的連結
+```
+
+![](/img/git-contribution-guide/AddUpstream.png)
+
+[GitHub Flow](https://medium.com/@trylovetom/%E8%AE%93%E6%88%91%E5%80%91%E4%BE%86%E4%BA%86%E8%A7%A3-github-flow-%E5%90%A7-4144caf1f1bf)、GitLab Flow 建議我們在多人實作時建立分支，目的在能夠安全實驗程式碼與執行變更，接下來，輸入以下指令建立新分支並切換過去：
+
+```shell showLineNumbers
+git checkout -b 分支名稱
+```
+
+![](/img/git-contribution-guide/ChangeToNewBranch.png)
+
+然後就可以開始編輯了！  
+
+![](/img/git-contribution-guide/StartContribution.png)
+
+當您每編輯到了一個階段後，您可以在該資料夾下開啟終端機，並 **適當** 使用下方指令進行將新增檔案至索引、儲存變更、提交推送變更至 Git 平台上的副本專案等動作：
+
+```shell showLineNumbers
+git add . # 將所有檔案新增至索引，如果沒有新檔案可以不用輸入這行
+git commit -am "更新內容" # 提交變更
+git push -u origin 剛剛建立的新分支名稱 # 推送變更至Git平台的副本專案
+```
+
+<Adsense/>
+
+![](/img/git-contribution-guide/AddAndCommit.png)
+
+## 步驟三：向原專案發出推送要求  
+
+請先前往 Git 平台上專案副本網頁，點選 `Contribute` -> `Open Pull Request`
+
+![](/img/git-contribution-guide/github/WhereOpenPRButton.png)
+
+:::info 小筆記
+
+* GitHub 到專案頁面，如果有出現 `XXX had recent pushes...` 也可以按下該黃色橫幅右側的 `Compare & pull request`：
+
+	![](/img/git-contribution-guide/CompareAndPRMessage.png)
+
+* GitLab 到專案頁面，在專案副本網頁上方會出現一則寫著 `You pushed to...` 的訊息，請點選下方的 `Create Merge Request` 按鈕：
+
+	![](/img/git-contribution-guide/gitlab/WhereOpenPRButton.png)
+
+:::
+
+接下來就會出現填寫 Pull Request 資訊的介面，填好相關資訊以後請按下 `Create Pull Request` 按鈕正式向原專案發出推送要求。
+
+![](/img/git-contribution-guide/WritePRInformation.png)
+
+創建完成之後將會透過 Email 通知該 Pull Request 的各種更新動態，
+![](/img/git-contribution-guide/WaitForMerge.png)  
+
+## 等待結果！
+
+當有人對您發出的 Pull Request 做出回應時，您會收到相關通知：
+
+<Adsense/>
+
+![](/img/git-contribution-guide/PRNotification.png)
+
+:::caution
+如果是 GitLab 則不會通知！請自己定時前往確認您發出的推送請求之目前處理進度！
+:::
+
+點進去即可查看目前該 Pull Request 的相關消息，如果您有看到 Pull Request 標題下方有 `Merged` 字眼，恭喜您，您的推送請求已經被開發者批准了！您可以按下下方的 `Delete branch` 刪除分支！
+
+![](/img/git-contribution-guide/PRMerged.png)
+
+## 繼續貢獻
+
+當然您可以選擇繼續貢獻，但在繼續貢獻之前，請檢察是否原專案的主分支有被變更過，如果有，請先輸入以下指令以拉取原專案有被做過的變更：
+
+```shell showLineNumbers
+git pull upstream master
+```
+
+:::caution 我想先告知你
+若您是 GitLab，主分支通常會是 `main` 而不是 `master`！不過可能依照每個專案會有不同的設定。
+:::
+
+然後您就可以繼續編輯等作業了
+
+## 常見問題
+### 無法使用 `git commit` ？
+#### `Please Tell me who you are`
+
+這通常會發生在您第一次使用 Git CLI 進行專案管理的時候，因為他沒有您的資訊，所以 `git commit` 指令無法執行，請跟著他的指令提示填入您的相關資訊後，再輸入一次 `git commit` 即可正常提交。
+
+<Adsense/>
+
+![](/img/git-contribution-guide/NoGlobalAccountError.png)
+
+#### `Nothing to commit, working tree clean`
+
+這代表您沒有對檔案做出變更，或是沒有使用 `git add` 指令先將檔案加入到 Git 索引內。
+
+![](/img/git-contribution-guide/CommitWorkingTreeClean.png)
+
+### 無法使用 `git push` ？
+#### `Everything up-to-date`
+
+這跟剛剛 `Nothing to commit, working tree clean` 差不多，您沒有檔案的修改需要被推送上 Git 平台。
+
+
+![](/img/git-contribution-guide/PushUpToDate.png)
+#### `Support for password autoentication was removed...`
```

Comment:
Maybe add gh (GitHub cli) to here?

Reply:
hmm no idea how to use gh cli, u wanna write that?